src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -370,10 +389,17 @@ private void EmitModule(JsonTextWriter memoryWriter, ModuleSymbol moduleSymbol,
             memoryWriter.WriteStartObject();
 
             var body = moduleSymbol.DeclaringModule.Value;
-            if (body is IfConditionSyntax ifCondition)
+            switch (body)
```

Comment:
[nit] commonize or use the same pattern (if/elseif or switch) in both module & resource processing functions

Reply:
Yeah should be doable.

---
In reply to: [582213424](https://github.com/Azure/bicep/pull/1521#discussion_r582213424) [](ancestors = 582213424)

Reply:
Done